pkg/apis/scheduling/v1alpha1/types_placement.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/apis/conditions/v1alpha1"
+)
+
+// placement defines a selection rule to choose ONE location for MULTIPLE namespaces in a workspace.
+//
+// placement is in Pending state initially, when a location is selected by the placement, the placement
+// turns to UnBound state. In Pending or UnBound state, the selection rule can be updated to select another location.
+// When the user annotates a namespace with the key of "scheudling.kcp.dev/placement", the namespace will
+// pick one placement, and this placement is transfered to Bound state. The spec of the placement is immutable
+// in Bound state. The placement will turns back to Unbound state when no namespace uses this
```

Comment:
have a little problems with marking it read-only. If this object comes from GitOps, that tool will see errors. We should rather make it mutable, but use a condition that says that the status is not up-to-datae.